tools/autograd/load_derivatives.py

```diff
@@ -144,15 +144,25 @@ def set_up_derivatives(defn_name, defn, declaration):
 
         # Set up the derivative information
         derivatives = []
+        not_derivatives = []
+        not_args_with_gradients = []
         for raw_names in sorted(defn.keys()):
             formula = defn[raw_names]
             names = split_names(raw_names)
-            derivatives.append(create_derivative(declaration, formula, names))
+            if formula.lower().strip() == 'not_differentiable':
+                derivative = create_derivative(declaration['arguments'], declaration['returns'],
+                                               declaration['name'], formula, names)
+                not_derivatives.append(derivative)
+                not_args_with_gradients += derivative['var_names']
+            else:
+                derivatives.append(create_derivative(declaration['arguments'], declaration['returns'],
+                                                     declaration['name'], formula, names))
+        args_with_gradients = list(filter(lambda x: x['name'] not in not_args_with_gradients, args_with_gradients))
```

Comment:
does this have any effect?  What is this supposed to catch?  If all the arguments don't have gradients?